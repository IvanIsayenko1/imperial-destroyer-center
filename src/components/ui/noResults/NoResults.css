.no-results {
  padding: 3rem;
  text-align: center;
  background: rgba(255, 59, 59, 0.1);
  border: 1px solid rgba(255, 59, 59, 0.2);
  border-radius: 8px;
}

.no-results-content {
  display: flex;
  flex-direction: column;
  align-items: center;
  gap: 1rem;
}

.no-results-text {
  color: var(--rebel-base);
  font-size: 1.125rem;
  margin: 0;
}

.no-results-button {
  padding: 0.5rem 1rem;
  background: rgba(255, 59, 59, 0.2);
  border: 1px solid rgba(255, 59, 59, 0.3);
  color: var(--rebel-base);
  border-radius: 4px;
  font-size: 0.875rem;
  cursor: pointer;
  transition: all 0.2s ease;
}

.no-results-button:hover {
  background: rgba(255, 59, 59, 0.3);
  box-shadow: 0 0 10px rgba(255, 59, 59, 0.2);
}
