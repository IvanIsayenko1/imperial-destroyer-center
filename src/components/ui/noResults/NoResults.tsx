import React from 'react';
import './NoResults.css';

interface NoResultsProps {
  searchTerm: string;
  onClearFilters: () => void;
}

export const NoResults: React.FC<NoResultsProps> = ({
  searchTerm,
  onClearFilters,
}) => {
  return (
    <div className="no-results">
      <div className="no-results-content">
        <p className="no-results-text">
          No planets found matching &quot;{searchTerm}&quot;
        </p>
        <button onClick={onClearFilters} className="no-results-button">
          Clear Filters
        </button>
      </div>
    </div>
  );
};
