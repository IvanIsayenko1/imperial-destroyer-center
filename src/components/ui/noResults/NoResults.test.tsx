import { render, screen, fireEvent } from "@testing-library/react";
import { NoResults } from "./NoResults";

describe("NoResults component", () => {
  it("should render the correct message with the search term", () => {
    const searchTerm = "Tatooine";
    render(<NoResults searchTerm={searchTerm} onClearFilters={() => {}} />);

    const message = screen.getByText(
      `No planets found matching "${searchTerm}"`
    );
    expect(message).toBeInTheDocument();
  });

  it("should call onClearFilters when the button is clicked", () => {
    const onClearFilters = vi.fn();
    render(<NoResults searchTerm="Tatooine" onClearFilters={onClearFilters} />);

    const button = screen.getByText("Clear Filters");
    fireEvent.click(button);

    expect(onClearFilters).toHaveBeenCalledTimes(1);
  });

  it("should match snapshot", () => {
    const { asFragment } = render(
      <NoResults searchTerm="Tatooine" onClearFilters={() => {}} />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
