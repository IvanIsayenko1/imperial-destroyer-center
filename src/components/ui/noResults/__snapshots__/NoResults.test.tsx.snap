// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NoResults component > should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="no-results"
  >
    <div
      class="no-results-content"
    >
      <p
        class="no-results-text"
      >
        No planets found matching "Tatooine"
      </p>
      <button
        class="no-results-button"
      >
        Clear Filters
      </button>
    </div>
  </div>
</DocumentFragment>
`;
